pkg/server/grpc/server.go

```diff
@@ -171,6 +180,15 @@ func (s *Server) applyServerCertificate() (*tls.Certificate, error) {
 	return &cert, nil
 }
 
+func (s *Server) authenticate(ctx context.Context) *user {
+	for _, authn := range s.authenticators {
+		if u := authn.authenticate(ctx); u != nil {
```

Comment:
I think you may want to be explicit about the order here. E.g., use client cert authentication if both cert and jwt are presented.

Reply:
Where/how do you want me to make it explicit? Adding comments to where the authenticators get initialized?

Reply:
Maybe make authenticators a map[int]authenticator ? The int index can be the authenticator type enum. And here you can do:
if s.authenticators[CertAuthenticator] != nil {
  s.authenticators[CertAuthenticator].authenticate(ctx)
  ...
}
if s.authenticators[JwtAuthenticator] != nil {
  s.authenticators[JwtAuthenticator].authenticate(ctx)
  ...
}

Reply:
hmm I feel this is a verbose way to implement the same logic. if you're worrying about confusing users, I believe clear documentation makes more sense than code change.

Reply:
Sure, doc/comment is definitely helpful. We can use the order of s.authenticators to specify the authenticator usage order. So now it is first cert then JWT, as can be seen from the New() func.